### PR TITLE
Move event_object_movement data to src

### PR DESCRIPTION
**Pull request by GriffinRichards**

### CODE REVIEW

File: src/data/field_effects/field_effect_objects.h [hunk 1]
```diff
@@ -0,0 +1,1306 @@
...
+
+const struct SpriteFrameImage gUnknown_83A535C[] = { obj_frame_tiles(gUnknown_8398048) };
+const struct SpriteFrameImage gUnknown_83A5364[] = { obj_frame_tiles(gUnknown_8398068) };
+const struct SpriteFrameImage gUnknown_83A536C[] = { obj_frame_tiles(gUnknown_83980A8) };
+const struct SpriteFrameImage gUnknown_83A5374[] = { obj_frame_tiles(gUnknown_8398128) };
+
+const struct SpriteTemplate gUnknown_83A537C =
+{
+    .tileTag = 65535,
+    .paletteTag = 65535,
```
Comment (jiangzhengwenjz) [inline src/data/field_effects/field_effect_objects.h]: `SPRITE_INVALID_TAG` for 65535/0xFFFF

File: src/data/object_events/movement_action_func_tables.h [hunk 2]
```diff
@@ -0,0 +1,1646 @@
...
+    sMovementActionFuncs_xA3,
+    sMovementActionFuncs_xA4,
+    sMovementActionFuncs_xA5,
+    sMovementActionFuncs_xA6,
+    sMovementActionFuncs_xA7,
+    sMovementActionFuncs_xA8,
+    sMovementActionFuncs_xA9,
+};
+
+bool8 (*const sMovementActionFuncs_x00[])(struct ObjectEvent *, struct Sprite *) = {
```
Comment (jiangzhengwenjz) [inline src/data/object_events/movement_action_func_tables.h]: `static`

File: src/data/object_events/movement_action_func_tables.h [hunk 3]
```diff
@@ -0,0 +1,1646 @@
...
+bool8 MovementAction_EmoteQuestionMark_Step0(struct ObjectEvent *, struct Sprite *);
+bool8 MovementAction_EmoteHeart_Step0(struct ObjectEvent *, struct Sprite *);
+bool8 do_double_excl_bubble(struct ObjectEvent *, struct Sprite *);
+bool8 do_smile_bubble(struct ObjectEvent *, struct Sprite *);
+bool8 MovementAction_SetFixedPriority_Step0(struct ObjectEvent *, struct Sprite *);
+bool8 MovementAction_ClearFixedPriority_Step0(struct ObjectEvent *, struct Sprite *);
+bool8 MovementAction_InitAffineAnim_Step0(struct ObjectEvent *, struct Sprite *);
+bool8 MovementAction_ClearAffineAnim_Step0(struct ObjectEvent *, struct Sprite *);
+
+bool8 (*const sMovementActionFuncs_x00[])(struct ObjectEvent *, struct Sprite *);
```
Comment (jiangzhengwenjz) [inline src/data/object_events/movement_action_func_tables.h]: `static`

File: src/data/object_events/movement_action_func_tables.h [hunk 4]
```diff
@@ -0,0 +1,1646 @@
+bool8 MovementActionFunc_x04_0(struct ObjectEvent *, struct Sprite *);
```
Comment (jiangzhengwenjz) [inline src/data/object_events/movement_action_func_tables.h]: Are the functions static?

File: src/data/object_events/movement_type_func_tables.h [hunk 5]
```diff
@@ -0,0 +1,614 @@
+bool8 MovementType_WanderAround_Step0(struct ObjectEvent *objectEvent, struct Sprite *sprite);
```
Comment (jiangzhengwenjz) [inline src/data/object_events/movement_type_func_tables.h]: `static`?

File: src/data/object_events/object_event_graphics_info.h [hunk 6]
```diff
@@ -0,0 +1,154 @@
+const struct ObjectEventGraphicsInfo gObjectEventGraphicsInfo_83A3BB0 = {0xFFFF, OBJ_EVENT_PAL_TAG_8, OBJ_EVENT_PAL_TAG_10, 512, 16, 32, 0, SHADOW_SIZE_M, FALSE, FALSE, TRACKS_FOOT, &gObjectEventBaseOam_16x32, gSubspriteTables_83A379C, gAnimTable_83A3470, gSpriteImages_83A00A0, gDummySpriteAffineAnimTable};
```
Comment (jiangzhengwenjz) [inline src/data/object_events/object_event_graphics_info.h]: `SPRITE_INVALID_TAG`

File: src/event_object_movement.c [hunk 7]
```diff
@@ -217,175 +180,1045 @@ static u8 setup##_callback(struct ObjectEvent *objectEvent, struct Sprite *sprit
...
+    MovementType_JogInPlace,
+    MovementType_JogInPlace,
+    MovementType_Invisible,
+    sub_8063298,
+    sub_80632BC,
+    sub_80632E0,
+    MovementType_WanderAroundDuplicate,
+};
+
+const u8 gRangedMovementTypes[NUM_FIELD_MAP_OBJECT_TEMPLATES] = {
```
Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `bool8`

File: src/event_object_movement.c [hunk 8]
```diff
@@ -217,175 +180,1045 @@ static u8 setup##_callback(struct ObjectEvent *objectEvent, struct Sprite *sprit
...
 
+#include "data/object_events/object_event_graphics_info_pointers.h"
+#include "data/field_effects/field_effect_object_template_pointers.h"
+#include "data/object_events/object_event_pic_tables.h"
+#include "data/object_events/object_event_anims.h"
+#include "data/object_events/base_oam.h"
+#include "data/object_events/object_event_subsprites.h"
+#include "data/object_events/object_event_graphics_info.h"
+
+const struct SpritePalette sObjectEventSpritePalettes[] = {
```
Comment (jiangzhengwenjz) [inline src/event_object_movement.c]: `static`

File: src/event_object_movement.c [hunk 9]
```diff
@@ -217,175 +180,1045 @@ static u8 setup##_callback(struct ObjectEvent *objectEvent, struct Sprite *sprit
...
+#include "data/field_effects/field_effect_objects.h"
+
+const s16 gMovementDelaysMedium[] = {32, 64,  96, 128};
+const s16 gMovementDelaysLong[] =   {32, 64, 128, 192};
+const s16 gMovementDelaysShort[] =  {32, 48,  64,  80};
+
+#include "data/object_events/movement_type_func_tables.h"
+
+const u8 gFaceDirectionAnimNums[] = {
+    [DIR_NONE]      = 0x00,
```
Comment (PikalaxALT) [inline src/event_object_movement.c]: Use the constant defines here

Comment (GriffinRichards) [reply src/event_object_movement.c]: Which constants?

File: src/event_object_movement.c [hunk 10]
```diff
@@ -217,175 +180,1045 @@ static u8 setup##_callback(struct ObjectEvent *objectEvent, struct Sprite *sprit
...
+    [DIR_NORTH]     = { 0, -1},
+    [DIR_WEST]      = {-1,  0},
+    [DIR_EAST]      = { 1,  0},
+    [DIR_SOUTHWEST] = {-1,  1},
+    [DIR_SOUTHEAST] = { 1,  1},
+    [DIR_NORTHWEST] = {-1, -1},
+    [DIR_NORTHEAST] = { 1, -1},
+};
+
+const u8 gFaceDirectionMovementActions[] = {
```
Comment (PikalaxALT) [inline src/event_object_movement.c]: Use the movement action constants here (931 and below)

Comment (GriffinRichards) [reply src/event_object_movement.c]: Resolved

**Pull request merged**